.github/workflows/dependency-update.yml

```diff
@@ -0,0 +1,310 @@
+name: Automated Dependency Updates
+
+on:
+  # Run weekly on Mondays at 9 AM UTC
+  schedule:
+    - cron: '0 9 * * 1'
+  
+  # Allow manual triggering
+  workflow_dispatch:
+    inputs:
+      force_update:
+        description: 'Force update even if no changes detected'
+        required: false
+        default: 'false'
+        type: boolean
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-dependencies:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.24.5'
+    
+    - name: Install jq
+      run: sudo apt-get update && sudo apt-get install -y jq
+    
+    - name: Install Task
+      run: |
+        sudo sh -c "$(curl --location https://taskfile.dev/install.sh)" -- -d -b /usr/local/bin
+    
+    - name: Check for MCP registry updates
+      id: check_updates
+      run: |
+        echo "Current version:"
+        grep "github.com/modelcontextprotocol/registry" go.mod
+        
+        echo "Checking for updates..."
+        go list -u -m github.com/modelcontextprotocol/registry
+        
+        # Get current and latest versions
+        CURRENT_VERSION=$(grep "github.com/modelcontextprotocol/registry" go.mod | awk '{print $2}')
+        LATEST_VERSION=$(go list -m -versions github.com/modelcontextprotocol/registry | awk '{print $NF}')
+        
+        echo "current_version=$CURRENT_VERSION" >> $GITHUB_OUTPUT
+        echo "latest_version=$LATEST_VERSION" >> $GITHUB_OUTPUT
+        
+        if [[ "$CURRENT_VERSION" != "$LATEST_VERSION" ]] || [[ "${{ inputs.force_update }}" == "true" ]]; then
+          echo "update_needed=true" >> $GITHUB_OUTPUT
+          echo "Updates available: $CURRENT_VERSION -> $LATEST_VERSION"
+        else
+          echo "update_needed=false" >> $GITHUB_OUTPUT
+          echo "No updates available"
+        fi
+    
+    - name: Update Go dependencies
+      if: steps.check_updates.outputs.update_needed == 'true'
+      run: |
+        echo "Updating MCP registry dependency..."
+        go get -u github.com/modelcontextprotocol/registry@latest
```

Comment:
Don't we get this from Renovate?

Reply:
The version bump yes, but this also updates the upstream schema reference to point to the same commit hash as the dependency (this way dependency/structs we import + the schema move in sync). I know we talked renovate can do similar stuff, but I haven't played with it. If you suggest this is possible I'm happy to replace this workflow in favour of renovate?